src/modules/SdnDiag.Common/public/Start-SdnDataCollection.ps1

```diff
@@ -205,27 +205,63 @@ function Start-SdnDataCollection {
                         }
 
                         "{0} with role {1} added for data collection" -f $object.Name, $object.Role | Trace-Output
-                        $dataCollectionNodes += $object
+                        [void]$dataCollectionNodes.Add($object)
                     }
                 }
             }
         }
 
-        if ($null -eq $dataCollectionNodes) {
+        if ($dataCollectionNodes.Count -eq 0) {
             throw New-Object System.NullReferenceException("No data nodes identified")
         }
 
+        # once we have identified the nodes, we need to validate WinRM connectivity to the nodes
+        # if we are running on PowerShell 7 or greater, we can leverage the -Parallel parameter
+        # to speed up the process
+        # if we are running on PowerShell 5.1, we will need to run the process in serial
+        # if we have any nodes that fail the WinRM connectivity test, we will remove them from the data collection
+        "Validating WinRM connectivity to {0}" -f ($dataCollectionNodes.Name -join ', ') | Trace-Output
+        $nodesToRemove = New-Object System.Collections.ArrayList
+        if ($PSVersionTable.PSVersion.Major -ge 7) {
+            $dataCollectionNodes | Foreach-Object -ThrottleLimit 10 -Parallel {
+                $tncResult = Test-NetConnection -ComputerName $_.Name -Port 5985 -InformationLevel Quiet
+                if (-NOT ($tncResult)) {
+                    [void]$nodesToRemove.Add($_)
+                }
+            }
+        }
+        else {
+            $dataCollectionNodes | ForEach-Object {
+                $tncResult = Test-NetConnection -ComputerName $_.Name -Port 5985 -InformationLevel Quiet
+                if (-NOT ($tncResult)) {
+                    [void]$nodesToRemove.Add($_)
+                }
+            }
+        }
```

Comment:
Would it be benifical if we put the codeblock after ForEach-Object to a share script block variable considering the code block by PS 7 and 5.1 is to execute same code?